test/integration_test/fafb_matrix_test.go

```diff
@@ -0,0 +1,352 @@
+// +build fafb
+
+package integrationtest
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sort"
+	"strconv"
+	"strings"
+	"testing"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
+	coreops "github.com/portworx/sched-ops/k8s/core"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+// node* is to be used in the Node section of the StorageCluster spec. node0 will select the
+// alphabetically 1st PX node, node1 will select the 2nd, and so on
+const (
+	nodeReplacePrefix = "replaceWithNodeNumber"
+
+	node0 = nodeReplacePrefix + "0"
+	node1 = nodeReplacePrefix + "1"
+	node2 = nodeReplacePrefix + "2"
+)
+
+var (
+	auto    = "auto"    // Used for things like auto-journal drives
+	size3   = "size=3"  // Recommended journal drive size, more than this is useless: https://docs.portworx.com/install-with-other/operate-and-maintain/performance-and-tuning/tuning/
+	size32  = "size=32" // Default kvdb size from spec gen
+	size50  = "size=50"
+	size100 = "size=100"
+	size200 = "size=200"
+
+	disks100 = []string{size100}
+	disks200 = []string{size200}
+
+	threeStorageNodes = uint32(3)
+
+	internalKVDB     = v1.KvdbSpec{Internal: true}
+	autoJournalDrive = &v1.CloudStorageSpec{
+		CloudStorageCommon: v1.CloudStorageCommon{
+			KvdbDeviceSpec:    &size32,
+			JournalDeviceSpec: &auto, // TODO: Once PWX-21364 is fixed, add this back in!
+			DeviceSpecs:       &disks100,
+		},
+	}
+
+	// simpleClusterSpec is a StorageCluster with internal KVDB and auto journal drives
+	simpleClusterSpec = v1.StorageClusterSpec{
+		Kvdb:         &internalKVDB,
+		CloudStorage: autoJournalDrive,
+	}
+)
+
+type TestrailCase struct {
```

Comment:
can we move this to a common util package? and make `BackendRequirements` a pointer so it could be nil? this can be used for other tests

Reply:
Definitely, I was actually thinking about something similar, I have a few ways I can make it more generic (I'll probably make a `TestrailCase` in common test utils, and then a `PureTestrailCase` that will compose that and add the Pure backend requirements).